hyperjumptech/monika
# Fix/issue 599

# Monika Pull Request (PR)  

## What feature/issue does this PR add  
1. Issue #599 

## How did you implement / how did you fix it  
1.  Update the pkg version. Seems like there is a glitch in the patched node base for macOS which was included in the previous pkg version. Hence, the error when running the stand alone monika binary.
2. Create the binary using node 14. Node 12 is nearing end of life, so let's use minimum 14. Not 16 yet because sqlite 3 doesn't seem to support node 16 yet at this moment.
3. Add the sqlite3 bindings in the `assets` key in package.json. With this, we don't need to manually copy node_sqlite3.node anymore. The latest pkg will automatically include it as an asset.

## How to test  
1. Use Node 14.19.1 (the latest 14): Run `nvm install 14` if you use nvm.
1. Run `npm run pkg`
2. Run `cd dist && ./monika`. Note, please run the generated monika with both a valid monika.yml and without any configuration where it should show error.

## Note

While this PR is supposed to fix the generation of macOS stand alone binary, it affects the windows and linux platform as well. So please run the commands in the How to Test section above in Linux and windows as well to make sure the binaries are generated correctly on the respective platform.
